src/parse_cmd.rs

```diff
@@ -0,0 +1,98 @@
+pub trait ToCmd<'a> {
+    fn to_cmd(&'a self) -> Vec<String>;
+}
+
+impl<'a> ToCmd<'a> for str {
+    fn to_cmd(&'a self) -> Vec<String> {
+        let mut args = Vec::new();
+        let mut current_arg = String::new();
+        let mut in_quote = Vec::new();
+
+        for c in self.chars() {
+            if in_quote.is_empty() && c.is_whitespace() {
+                args.push(current_arg);
+                current_arg = String::new();
+                continue;
+            }
+
+            current_arg.push(c);
+
+            if c == '"' || c == '\'' {
+                if in_quote.last() == Some(&c) {
+                    in_quote.pop();
+                } else {
+                    in_quote.push(c);
+                }
+            }
```

Comment:
> `r#"echo "lorem' ipsum" dolor "sit' amet" '"#.to_cmd()`

_Great_ test case! I was so focused on getting other stuff that I think I actually went a bit overboard with this: We don't really have to deal with nested quotes at all – just recognize and ignore escaped quotes, right?

I've also just seen [this crate](https://github.com/comex/rust-shlex/blob/95ef6961a2500d89bc065b2873ca3e77850539e3/src/lib.rs), but I think it's more complex than we need here. I don't want to emulate shell syntax itself, just split some args :)

- - -

I think I can add quickcheck to test the invariant, but I'm not sure `JSON::from_str(stringify!("..."))` is that reliable. We'll see :)